### PR TITLE
Add support for hassubset and hassubsequence built-in functions

### PR DESCRIPTION
**Pull request by xuzhg**
Add support for hassubset and hassubsequence built-in functions

- Add UnboundFunctionHassubset and UnboundFunctionHassubsequence constants to ExpressionConstants
- Register both functions in UnboundFunctionNames to bypass standard built-in function lookup
- Implement ValidateArgsForCollection to enforce exactly two collection-valued arguments
- Return non-nullable Edm.Boolean as the result type for both functions
- Add functional tests in FilterAndOrderByFunctionalTests for filter expressions using hassubset and hassubsequence with collection properties and bracket collection literals
- Add binder unit tests in FunctionCallBinderTests covering happy paths, return type, empty collections, wrong argument count, and non-collection argument errors

### CODE REVIEW

File: test/UnitTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs [hunk 1]
```diff
@@ -3765,9 +3813,69 @@ public void FilterWithInOperationWithOpenTypesInMultipleEmptyStrings(string filt
...
+        [Fact]
+        public void CaseFunctionWithBooleanConditionsInFilter()
+        {
+            FilterClause filter = ParseFilter("case(ID gt 0:true, ID eq 0:false) eq true",
+                HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType(), HardCodedTestModel.GetPeopleSet());
+
+            BinaryOperatorNode binaryNode = filter.Expression.ShouldBeBinaryOperatorNode(BinaryOperatorKind.Equal);
+            SingleValueFunctionCallNode caseNode = binaryNode.Left.ShouldBeSingleValueFunctionCallQueryNode("case");
+            Assert.Equal(4, caseNode.Parameters.Count());
+        }
```
[the remark this answers is not in the document]
Comment (xuzhg) [reply test/UnitTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FilterAndOrderByFunctionalTests.cs:3827]: It's ok. Won't fix
